### ISSUE TITLE
CPU Load goes constant 100% if Proxy and Forward Proxy port match

### ISSUE THREAD

**Issue by wwwjj2** [shadowsocks/shadowsocks-windows#3398]
### Describe the bug
If general proxy port and Forward proxy port match, CPU goes constant 100% Load just after Proxy is accessed by any app

### Environment

- Shadowsocks client version: 4.4.1.0
- OS version: Windows 10 Pro 22H2 19045.2728
- .NET version: 4.8.04084

### Steps you have tried
1. General profile proxy (connected SS server local proxy port) is set to 1080
2. Forward Proxy Port is set to 1080, IP is set to 127.0.0.1 - 
![image](https://user-images.githubusercontent.com/61093153/231069115-262ad2bd-ac5c-43e4-9c6b-5946d8ae8ea2.png)
3. Enable Proxy SwitchyOmega Chrome Extension connected to SOCKS5 127.0.0.1:1080 - 
![image](https://user-images.githubusercontent.com/61093153/231069295-f6d317c9-67db-4d10-8310-ce6e7616425d.png)
4. Open any website

### What did you expect to see?
Anything but constant 100% CPU Load

### What did you see instead?
CPU Load goes 100% and never stops:
![image](https://user-images.githubusercontent.com/61093153/231068165-61ab8be6-e925-4060-8458-c3a6e7fcdd6c.png)
![image](https://user-images.githubusercontent.com/61093153/231068182-8034dd20-664e-4ffc-a3fd-bf2767e4eae8.png)
![image](https://user-images.githubusercontent.com/61093153/231068493-31a0a291-4c80-48da-9931-f8850ee5bffd.png)

### Config and error log in detail (with all sensitive info masked)

```
2023-04-11 08:52:00.5040|INFO|Shadowsocks.Controller.ShadowsocksController|WPF Localization Extension|Current culture: en-US
2023-04-11 08:52:00.6373|INFO|Shadowsocks.Controller.Listener|Shadowsocks started (4.4.1.0)
2023-04-11 08:52:33.6070|INFO|Shadowsocks.Controller.ShadowsocksController|WPF Localization Extension|Current culture: en-US
2023-04-11 08:52:34.2140|INFO|Shadowsocks.Controller.Listener|Shadowsocks started (4.4.1.0)
```
That's all what's in logs

**Comment by chenshaoju** [comment]
`Forward Proxy` is used for a few networks that need HTTP or SOCKS5 proxy to access the Internet.

If your network does not need the Proxy to use the internet, do not use `Forward Proxy`.

**Comment by wwwjj2** [comment]
> `Forward Proxy` is used for a few networks that need HTTP or SOCKS5 proxy to access the Internet.
> 
> If your network does not need the Proxy to use the internet, do not use `Forward Proxy`.

Yeah, right. I just thought this thing should never happen and should be fixed
